File: Client/Assets/Scripts/View/LoginWnd.cs

using UnityEngine;
using UnityEngine.UI;
using account;

public class LoginWnd : BaseWnd
{
    private InputField _account;

    private InputField _pwd;


    public void Initialize()
    {
        Button btnLogin = _transform.Find("BtnLogin").GetComponent<Button>();
        btnLogin.onClick.AddListener(OnBtnLoginClick);

        Button btnRegister = _transform.Find("BtnRegister").GetComponent<Button>();
        btnRegister.onClick.AddListener(OnBtnRegisterClick);

        _account = _transform.Find("Acc").GetComponent<InputField>();
        _pwd = _transform.Find("Pwd").GetComponent<InputField>();
    }

    private void OnBtnLoginClick()
    {
        ReqLogin req = new ReqLogin();


        //req.account = _account.text;
        //req.password = _pwd.text;
        //Debug.Log(string.Format("{0},{1}", req.account, req.password));
        //Net.instance.Send((int)MsgID.ACC_LOGIN_CREQ, req);
    }

    private void OnBtnRegisterClick()
    {
        if(string.IsNullOrEmpty(_account.text) || string.IsNullOrEmpty(_pwd.text))
        {
            MessageBox.Show("账号或者密码不能为空");
            return;
        }

        if(_account.text.Contains("'\'") || _account.text.Contains("'"))
        {
            MessageBox.Show("账号或者密码包含非法字符");
            return;
        }

        ReqRegister req = new ReqRegister();
        req.account = _account.text;
        //req.password = _pwd.text;

        Net.instance.Send((int)MsgID.ACC_REG_CREQ, req);
    }
}

File: Client/Assets/StreamingAssets/lua/View/LoginWnd.lua

require "Common/define"
require "3rd/pbc/protobuf"
require "3rd/pblua/account_pb"


LoginWnd = {};
local this = LoginWnd;

--local panel;
local behavior;
local transform;
local gameObject;
local btnLogin;
local btnRegister;
local acc;
local pwd;
--构建函数--
function LoginWnd.New()
	logWarn("LoginWnd.New--->>");

	return this;
end

function LoginWnd.Open()
	logWarn("LoginWnd.Awake--->>");
	panelMgr:CreatePanel('LoginWnd', this.OnCreate);
end

--启动事件--
function LoginWnd.OnCreate(obj)

	gameObject = obj;
	transform = obj.transform;
--[[
	this.btnOpen = transform:Find("Open").gameObject;
	this.gridParent = transform:Find('ScrollView/Grid');
--]]
	btnLogin = transform:FindChild("BtnLogin").gameObject;
	btnRegister = transform:FindChild("BtnRegister").gameObject;

	acc = transform:FindChild("Acc"):GetComponent('InputField');
	pwd = transform:FindChild("Pwd"):GetComponent('InputField');

	--panel = transform:GetComponent('UIPanel');
	behavior = transform:GetComponent('LuaBehaviour');
	--logWarn("Start lua--->>"..gameObject.name);

	behavior:AddClick(btnLogin, this.OnLoginClick);
	behavior:AddClick(btnRegister, this.OnRegisterClick);
	--resMgr:LoadPrefab('PromptItem', { 'PromptItem' }, this.InitPanel);

end

--初始化面板--
function LoginWnd.InitPanel(objs)
	local count = 100; 
	local parent = this.gridParent;
	for i = 1, count do
		local go = newObject(objs[0]);
		go.name = 'Item'..tostring(i);
		go.transform:SetParent(parent);
		go.transform.localScale = Vector3.one;
		go.transform.localPosition = Vector3.zero;
		behavior:AddClick(go, this.OnItemClick);

	    local label = go.transform:Find('Text');
	    label:GetComponent('Text').text = tostring(i);
	end
end

--滚动项单击--
function LoginWnd.OnItemClick(go)
    log(go.name);
end

--登录事件--
function LoginWnd.OnLoginClick(go)

	local req = account_pb.ReqLogin();
	req.account = acc.text;
	req.password = pwd.text;

	local msg = req:SerializeToString();

	local buffer = ByteArray.New();
	buffer:write(msg);
	networkMgr:SendMessage(1001, buffer);
    logWarn("OnClick---->>>"..go.name);
end

--注册事件--
function LoginWnd.OnRegisterClick(go)
	local req = account_pb.ReqRegister();
	req.account = acc.text;
	req.password = pwd.text;
	local msg = req:SerializeToString();

	local buffer = ByteArray.New();
	buffer:write(msg);
	networkMgr:SendMessage(1003, buffer);
end

--关闭事件--
function LoginWnd.Close()
	panelMgr:ClosePanel(CtrlNames.Prompt);
end